wallet/src/utils/Cryptid/cryptid.tsx

```diff
@@ -161,6 +161,89 @@ export class CryptidAccount {
 
     return nativeTransfer(this.connection, signingWrapper, destination, amount);
   };
+
+  getTokenAccountInfo = async (): Promise<{ publicKey: PublicKey, parsed: TokenInfo }[]> => {
+    let accounts: {
+      publicKey: PublicKey,
+      accountInfo: TokenAccountInfo,
+    }[] = this.address ? await getOwnedTokenAccounts(this.connection, await this.address) : [];
+    return accounts.map<{
+      publicKey: PublicKey,
+      parsed: TokenInfo,
+    }>(({ publicKey, accountInfo }) => {
+      setInitialAccountInfo(this.connection, publicKey, accountInfo);
+      return {
+        publicKey,
+        parsed: parseTokenAccountData(accountInfo.data),
+      };
+    }).sort((account1, account2) =>
+        account1.parsed.mint.toBase58().localeCompare(account2.parsed.mint.toBase58())
+    );
+  }
+}
+
+export function useCryptidWalletPublicKeys(cryptid: CryptidAccount | null): [PublicKey[], boolean] {
+  let [tokenAccountInfo, loaded] = useAsyncData(
+      cryptid ? cryptid.getTokenAccountInfo : async () => [],
+      cryptid ? cryptid.getTokenAccountInfo : async () => [], //TODO: Is thsi the best way to handle null?
+  );
+  let publicKeys = [
+      ...(cryptid && cryptid.address ? [cryptid.address] : []),
+      ...(tokenAccountInfo ? tokenAccountInfo.map(({ publicKey }) => publicKey) : []),
+  ]
+  publicKeys = useRefEqual(
+      publicKeys,
+      (oldKeys, newKeys) =>
+          oldKeys.length === newKeys.length
+          && oldKeys.every((key, i) => key.equals(newKeys[i]))
+  );
+  return [publicKeys, loaded]
+}
+
+export async function signAndSendCryptidTransaction(
```

Comment:
Was is required to introduce that? What was `signAndSendTransaction` not working (which we used e.g. for native transfer?)

Reply:
I split it to add types to `signAndSendTransaction`, if we don't care about typing it I can make the original function work

Reply:
I just only want to have one. So if you add one I think I'd ask you to take out the original one in js and reference your new one. Otherwise it's confusing for the next person.

Reply:
Im missing some context but I think I agree with @rado0x54 here.